sips/sip-58.md

```diff
@@ -0,0 +1,274 @@
+During checkpoint construction, all accumulator events emitted by the transactions in the checkpoint are merged together, and one or more settlement transactions are created.
+Settlement transactions are system transactions which update the values of each accumulator written to by the transactions in the checkpoint.
+
+The Address Balance system is built on the accumulator system, with the following additional properties:
+- Any transaction can merge (deposit) to any accumulator.
+- Only the owner of an accumulator can split (withdraw) from it.
+- `T` must be a type of the form `0x2::balance::Balance<S>`, that is, only standard currencies can be sent using address balances.
+
+The accumulator system is not currently extensible by user code, but it may be in the future if there are compelling use cases for this.
+
+
+### Transaction Format
+
+Transactions gain a new type of `CallArg` called `CallArg::FundsWithdrawal` which expresses an attempted reservation from an address balance.
+`CallArg::FundsWithdrawal` specifies the amount of the reservation, the type of currency being reserved, and the owner of the funds (typically the transaction sender).
+
+At execution time, all `CallArg::FundsWithdrawal` inputs are converted into `0x2::funds_accumulator::Withdrawal<T>` objects.
+These objects behave similarly to `0x2::balance::Balance<T>` in that they have `split` and `join` operations.
+They can be converted into `0x2::balance::Balance<T>` via the `redeem()` method.
+Withdrawals are not actually considered to have been withdrawn unless redeemed.
+In other words, if you create a `Withdrawal<T>` but do not `redeem()` it, your balance will not change.
+
+The purpose of the reservation system is two-fold:
+- It allows signers of transactions to easily verify the maximum amount of outflows from their account before signing a transaction.
+- It allows the scheduler to avoid scheduling a transaction or set of transactions that could cause underflow of any account. Because the scheduler guarantees that underflow cannot occur, the system does not require any account-level locking at execution time for safety. This results in unlimited parallelism for withdrawals.
+
+### Move API
+
+The Move API consists of the following three functions:
+
+        /// Send a `Balance` to an address's funds accumulator.
+        public fun send_funds<T>(balance: Balance<T>, recipient: address) {
+            sui::funds_accumulator::add_impl(balance, recipient);
+        }
+
+        /// Redeem a `Withdrawal<Balance<T>>` to get the underlying `Balance<T>` from an address's funds
+        /// accumulator.
+        public fun redeem_funds<T>(withdrawal: sui::funds_accumulator::Withdrawal<Balance<T>>): Balance<T> {
+            withdrawal.redeem()
+        }
+
+        /// Create a `Withdrawal<Balance<T>>` from an object to withdraw funds from it.
+        public(package) fun withdraw_funds_from_object<T>(
```

Comment:
Does it make sense to show a public API instead?

Reply:
oh this will be public soon. will fix